src/tests/functional/frame_tests.ts

```diff
@@ -10,6 +10,38 @@ export class FrameTests extends FunctionalTestCase {
     await this.nextBeat
     this.assert.notOk(await this.innerHTMLForSelector("#missing"))
   }
+
+  async "test following a link within a frame with a target set navigates the target frame"() {
+    await this.clickSelector("#hello a")
+    await this.nextBeat
+
+    const frameText = await this.querySelector("#frame h2")
+    this.assert.equal("Frame: Loaded", await frameText.getVisibleText())
+  }
+
+  async "test following a link within a frame whose ancestor declares a target set navigates the target frame"() {
```

Comment:
I'm not totally clear on whether or not this is what we want.

Is each `<turbo-frame>` element to be treated as its own navigation context, or does it inherit from its ancestors?

According to the [MDN documentation for `<a target="...">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-target), the default value for `[target]` when omitted is `_self` (i.e. "the current browsing context").

If we _do_ want descendant frames to inherit the `[target]` of their closest ancestor, we have some options:

```html
<!-- inherit from ancestor -->
<turbo-frame id="parent" target="_top">
  <turbo-frame id="child"> <!-- inherits target="_top" -->
    <a href="/some-other-page.html">Navigates page</a>
  </turbo-frame>
</turbo-frame>

<!-- inherit from ancestor -->
<turbo-frame id="parent" target="_top">
  <turbo-frame id="child" target="child"> <!-- overrides target to itself -->
    <a href="/some-other-page.html">Navigates #child</a>
  </turbo-frame>
</turbo-frame>
```

This also might be an opportunity to introduce `_self` or `_parent` as special case values, similar to how we treat `_top`.

```html
<!-- each frame as self-contained context -->
<turbo-frame id="parent">
  <turbo-frame id="child" target="_self"><!-- the default -->
    <a href="/some-other-page.html">Navigates #child</a>
  </turbo-frame>

  <!- ... -->
  <turbo-frame id="child" target="_parent">
    <a href="/some-other-page.html">Navigates #parent</a>
  </turbo-frame>

  <!- ... -->
  <turbo-frame id="child" target="_top">
    <a href="/some-other-page.html">Navigates the page</a>
  </turbo-frame>
</turbo-frame>
```

If we'd still like to decide separate from the bug fix for #115, we can omit this test case and replace 

```diff
- const id = element.getAttribute("data-turbo-frame")
+ const id = element.getAttribute("data-turbo-frame") || this.element.closest("turbo-frame[target]")?.getAttribute("target") || ""
```

with 

```diff
- const id = element.getAttribute("data-turbo-frame")
+ const id = element.getAttribute("data-turbo-frame") || this.element.getAttribute("target") || ""
```

Reply:
Instead of defaulting to the concept of `_parent`, default to `_self`.